### PR TITLE
Improve tests by making them all have useful assertions

### PR DESCRIPTION
**Pull request by MarcelTorne**
In this PR we are improving the tests codebase so that all of the tests have useful assertions.

### CODE REVIEW

**Comment by aman3014** [comment]
> fun createResult(): Result {
        val request: Request = createRequest()
        return Result(request, Result.Code.SUCCESS, createToken(request.permissions), null, null)
    }

I saw that this function in LoginClientCreator is only used in testing, should we move it to the test itself then?

**Comment by MarcelTorne** [comment]
> > fun createResult(): Result {
> > val request: Request = createRequest()
> > return Result(request, Result.Code.SUCCESS, createToken(request.permissions), null, null)
> > }
> 
> I saw that this function in LoginClientCreator is only used in testing, should we move it to the test itself then?

@tberkane and @ahmedc3ca ?

File: app/src/main/java/com/github/epfl/meili/poi/PoiGoogleRetriever.kt [hunk 1]
```diff
@@ -113,7 +113,7 @@ data class GooglePlacesResponse(
     }
 }
 
-//TODO: we could also get some photos from the GooglePlaces API if we want
+//Attention: we could also get some photos from the GooglePlaces API if we want
```
Comment (aman3014) [inline app/src/main/java/com/github/epfl/meili/poi/PoiGoogleRetriever.kt:116]: should we just remove this comment?

File: app/src/main/java/com/github/epfl/meili/util/RotationGestureDetector.kt [hunk 2]
```diff
@@ -5,7 +5,7 @@ import kotlin.math.atan2
 
 // Reference used : https://stackoverflow.com/questions/10682019/android-two-finger-rotation
 /** Detects two finger rotation motions */
-class RotationGestureDetector(// Object which can be rotated using two fingers
+open class RotationGestureDetector(// Object which can be rotated using two fingers
```
Comment (aman3014) [inline app/src/main/java/com/github/epfl/meili/util/RotationGestureDetector.kt:8]: Why is this class being made open?

File: app/src/main/java/com/github/epfl/meili/util/RotationGestureDetector.kt [hunk 3]
```diff
@@ -31,7 +31,7 @@ class RotationGestureDetector(// Object which can be rotated using two fingers
     }
 
     /** Computes correct indices and positions based on the touch event */
-    fun onTouchEvent(event: MotionEvent): Boolean {
+    open fun onTouchEvent(event: MotionEvent): Boolean {
```
Comment (aman3014) [inline app/src/main/java/com/github/epfl/meili/util/RotationGestureDetector.kt:34]: Same question for this function

**Comment by aman3014** [comment]
> Actually this file is already in the test folder. Do you think it is good enough?

Yes! I missed that.

File: app/src/main/java/com/github/epfl/meili/util/RotationGestureDetector.kt (hunk 2 above)
Comment (MarcelTorne) [reply app/src/main/java/com/github/epfl/meili/util/RotationGestureDetector.kt:8]: It's a leftover from when I was testing the class, I will remove it.

**Review by aman3014 — APPROVED** [review]
Thanks!

**Review by Bwananah — APPROVED** [review]
Looks good ! Well done finding a way to test part of the rotation gesture detector ! Thanks !

**Pull request merged**